app/controllers/jobseekers/job_applications_controller.rb

```diff
@@ -13,7 +13,8 @@ def create
   end
 
   def submit
-    return redirect_to expired_jobseekers_job_job_application_path(vacancy.id) unless vacancy.listed?
+    raise ActionController::RoutingError, "Cannot submit application for non-listed job" unless vacancy.listed?
```

Comment:
This line won't be run when a vacancy is not listed, because the previous line returns.

Reply:
Yeah you've lost me here 😂

Reply:
I didn't notice that the return line was removed, not added!

Reply:
Need to adjust contrast on my monitor...